### PR TITLE
Cleanup and extensions

### PR DESCRIPTION
**Pull request by OlivierBondu**
Now runPostCrab should handle injecting in the DB an 'extended' sample if you pass more than 1 config file as argument:
- each crab task should preexist in the DB first, and inherit from the same process (aka: the first part of the dataset name should be the same)
- the name will be of the type `process_extended_i_plus_j_tags` where `i` and `j` are the original sample IDs
- the `sumw` and `nevents` are summed, an 'or' of the `processed_lumi` jsons is done

The PR also include the cleanup from the fact datasets are now bookkept in the [Datasets](https://github.com/cp3-llbb/Datasets) repo

### CODE REVIEW

**Comment by OlivierBondu** [comment]
Ah sorry, forgot to add: this was tested and injection seems successful:

```
>search_SAMADhi.py sample -n *v2.0.4+7415_HHAnalysis_2016-02-14.v0 | grep ggZH_HToBB_ZToNuNu
1206    ggZH_HToBB_ZToNuNu_M125_13TeV_powheg_pythia8_v2.0.4+7415_HHAnalysis_2016-02-14.v0
1255    ggZH_HToBB_ZToNuNu_M125_13TeV_powheg_pythia8_ext1_v2.0.4+7415_HHAnalysis_2016-02-14.v0
1313    ggZH_HToBB_ZToNuNu_M125_13TeV_powheg_pythia8_extended_1206_plus_1255_v2.0.4+7415_HHAnalysis_2016-02-14.v0
```

**Comment by OlivierBondu** [comment]
PR updated with @blinkseb comments : the tool has been moved to its own script for clarity, current naming is `mergeDBsamples`, but better ideas are welcome !

File: scripts/mergeDBSamples.py [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+
+# import CRAB3 stuff
+from CRABAPI.RawCommand import crabCommand
+# import some lumi utils
+from FWCore.PythonUtilities.LumiList import LumiList
+# import a bit of ROOT
+import ROOT
+ROOT.gROOT.Reset()
+
+def get_file_data(pfn):
```
Comment (blinkseb) [inline scripts/mergeDBSamples.py]: Not used anywhere

File: scripts/mergeDBSamples.py [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+# Add default ingrid storm package
+sys.path.append('/nfs/soft/python/python-2.7.5-sl6_amd64_gcc44/lib/python2.7/site-packages/storm-0.20-py2.7-linux-x86_64.egg')
+sys.path.append('/nfs/soft/python/python-2.7.5-sl6_amd64_gcc44/lib/python2.7/site-packages/MySQL_python-1.2.3-py2.7-linux-x86_64.egg')
+
+from SAMADhi import Dataset, Sample, File, DbStore
+import das_import
+
+# import CRAB3 stuff
+from CRABAPI.RawCommand import crabCommand
```
Comment (blinkseb) [inline scripts/mergeDBSamples.py]: Not needed

File: scripts/mergeDBSamples.py [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+                extras_sumw[suffix] = obj.GetVal()
+
+    entries = None
+    tree = f.Get("t")
+    if tree:
+        entries = tree.GetEntriesFast()
+
+    return (nominal_sumw, extras_sumw, entries)
+
+def sum_dicts(a, b):
```
Comment (blinkseb) [inline scripts/mergeDBSamples.py]: Not used

**Comment by blinkseb** [comment]
Great thanks! I'm wondering also if it would not be easier / prettier to pass sample ids to `mergeDBsamples` instead of crab tasks? Indeed, it has no links anymore with crab, since it's only merging two database entries into one. What do you think?

**Comment by OlivierBondu** [comment]
yup, can be done, the only slightly troublesome point would be to find back the tags from the urls stored in `code_version`, but definitely not a show stopper...

**Comment by blinkseb** [comment]
If it's finish, would it be possible to squash the 7 last commits into one?

**Comment by OlivierBondu** [comment]
I'm still adding the option to pass SAMADhi IDs instead of crab configs, but then yes, I can squash

**Comment by blinkseb** [comment]
Investigating a possible issue in extra sum of weights invalid json

File: scripts/mergeDBSamples.py [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+        for lfn, pfn, event_weight_sum, file_extras_event_weight_sum, nevents in list(results.values(File.lfn, File.pfn, File.event_weight_sum, File.extras_event_weight_sum, File.nevents)):
+            f = File(lfn, pfn, event_weight_sum, file_extras_event_weight_sum, nevents)
+            sample.files.add(f)
+        # Get info from parent datasets
+        results = dbstore.find(Dataset, Dataset.dataset_id == s['dataset_id'])
+        dataset_nevents +=  results[0].nevents
+    if len(extras_event_weight_sum) > 0:
+        sample.extras_event_weight_sum = unicode(extras_event_weight_sum)
+    if len(processed_lumi.getCompactList()) > 0:
+        sample.processed_lumi = unicode(processed_lumi.getCompactList())
```
Comment (blinkseb) [inline scripts/mergeDBSamples.py]: This must be encoded into JSON, a python representation is not valid. It should be:

``` python
unicode(json.dumps(processed_lumi.getCompactList()))
```

**Comment by OlivierBondu** [comment]
Now rebased (except for the two 'cleaning' commits') !

**Comment by OlivierBondu** [comment]
Can we remove the 'do not merge' ? Or is there still wanted features ?

**Comment by blinkseb** [comment]
Ah sorry I forgot it's was me who added it ...

**Pull request merged**